SIL.Windows.Forms.Keyboarding/Linux/GnomeShellIbusKeyboardSwitchingAdaptor.cs

```diff
@@ -3,22 +3,29 @@
 
 using System;
 using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
 using SIL.Reporting;
 
 namespace SIL.Windows.Forms.Keyboarding.Linux
 {
 	/// <summary>
 	/// Class for dealing with ibus keyboards on Gnome Shell (as found in Ubuntu Bionic >= 18.04)
 	/// </summary>
-	public class GnomeShellIbusKeyboardSwitchingAdaptor: UnityIbusKeyboardSwitchingAdaptor, IUnityKeyboardSwitchingAdaptor
+	public class GnomeShellIbusKeyboardSwitchingAdaptor : IbusKeyboardSwitchingAdaptor
 	{
+		private KeyboardDescription _defaultKeyboard;
+
 		public GnomeShellIbusKeyboardSwitchingAdaptor(IIbusCommunicator ibusCommunicator)
 			: base(ibusCommunicator)
 		{
 		}
 
-		void IUnityKeyboardSwitchingAdaptor.SelectKeyboard(uint index)
+		protected override void SelectKeyboard(KeyboardDescription keyboard)
 		{
+			var ibusKeyboard = (IbusKeyboardDescription) keyboard;
```

Comment:
This cast looks like it could fail. Do we know it will always be an object of class IbusKeyboardDescription?